src/test/regress/sql/view_propagation.sql

```diff
@@ -497,6 +497,30 @@ DROP VIEW IF EXISTS axx.temp_view_to_drop;
 DROP MATERIALIZED VIEW IF EXISTS axx.temp_view_to_drop;
 DROP MATERIALIZED VIEW IF EXISTS axx.temp_view_to_drop;
 
+-- Test the GUC citus.local_view_dependency_checks
+SET search_path to view_prop_schema;
+
+CREATE TABLE dist_t (a int);
+SELECT create_distributed_table('dist_t', 'a');
+
+CREATE TABLE local_t (a int);
+
+SET citus.local_view_dependency_checks TO false;
+-- views will be created locally with warnings
+CREATE VIEW vv1 as SELECT * FROM local_t;
+CREATE OR REPLACE VIEW vv2 as SELECT * FROM vv1;
+-- view will fail due to local circular dependency
+CREATE OR REPLACE VIEW vv1 as SELECT * FROM vv2;
+
+SET citus.local_view_dependency_checks TO true;
+SET citus.enable_unsupported_feature_messages TO false;
+-- views will be created locally without errors OR warnings
+CREATE VIEW vv3 as SELECT * FROM local_t;
+CREATE OR REPLACE VIEW vv4 as SELECT * FROM vv3;
+CREATE OR REPLACE VIEW vv3 as SELECT * FROM vv4;
```

Comment:
normally, one would expect that create_dist_table on `local_t` should fail, because it has a circular dependency. What happens is that we simply skip those views.

But, we at least do not allow creating it back. So, can you add:

```SQL
SELECT create_distributed_table('local', 'a');
-- show that views are NOT in pg_dist_object
...
-- show that we cannot re-create the circular views ever
SET citus.enable_unsupported_feature_messages TO false;
 CREATE OR REPLACE VIEW vv1 as SELECT * FROM vv2;
ERROR:  Citus can not handle circular dependencies between distributed objects
DETAIL:  "view vv1" circularly depends itself, resolve circular dependency first
SET citus.enable_unsupported_feature_messages TO true;
 CREATE OR REPLACE VIEW vv1 as SELECT * FROM vv2;
ERROR:  Citus can not handle circular dependencies between distributed objects
DETAIL:  "view vv1" circularly depends itself, resolve circular dependency first
```